test/jdk/java/lang/Thread/UncaughtExceptionsTest.java

```diff
@@ -0,0 +1,206 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static java.lang.System.err;
+import static java.lang.System.out;
+
+/*
+ * @test
+ * @bug 4833089 4992454
+ * @summary Check for proper handling of uncaught exceptions
+ * @author Martin Buchholz
+ * @library /test/lib
+ * @build jdk.test.lib.process.*
+ * @run testng UncaughtExceptionsTest
+ */
+public class UncaughtExceptionsTest {
+
+    @DataProvider
+    public Object[][] testCases() {
+        return new Object[][]{
+            new Object[] { "ThreadIsDeadAfterJoin",
+                           0,
+                           Seppuku.EXPECTED_RESULT,
+                           "Exception in thread \"Thread-0\".*Seppuku"
+            },
+            new Object[] {
+                            "MainThreadAbruptTermination",
+                            1,
+                            Seppuku.EXPECTED_RESULT,
+                            "Exception in thread \"main\".*Seppuku"
+            },
+            new Object[] { "MainThreadNormalTermination", 0, Seppuku.EXPECTED_RESULT, ""},
+            new Object[] { "DefaultUncaughtExceptionHandlerOnMainThread", 1, Seppuku.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnMainThreadOverride", 1, Seppuku.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnNonMainThreadOverride", 0, Seppuku.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnNonMainThread", 0, Seppuku.EXPECTED_RESULT, "" },
+            new Object[] { "ThreadGroupUncaughtExceptionHandlerOnNonMainThread", 0, Seppuku.EXPECTED_RESULT, "" }
+        };
+    }
+
+    @Test(dataProvider = "testCases")
+    public void test(String className, int exitValue, String stdOutMatch, String stdErrMatch) throws Throwable {
+        ProcessBuilder processBuilder = ProcessTools.createJavaProcessBuilder(String.format("Seppuku$%s",className));
+        OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(processBuilder);
+        outputAnalyzer.shouldHaveExitValue(exitValue);
+        outputAnalyzer.stderrShouldMatch(stdErrMatch);
+        outputAnalyzer.stdoutShouldMatch(stdOutMatch);
+    }
+
+}
+
+
+class OK implements Thread.UncaughtExceptionHandler {
+    public void uncaughtException(Thread t, Throwable e) {
+        out.println(Seppuku.EXPECTED_RESULT);
+    }
+}
+
+class NeverInvoked implements Thread.UncaughtExceptionHandler {
+    public void uncaughtException(Thread t, Throwable e) {
+        err.println("Test failure: This handler should never be invoked!");
+    }
+}
+
+class Seppuku extends Thread implements Runnable {
+
+    final static String EXPECTED_RESULT = "OK";
+
+    public static void seppuku() { throw new RuntimeException("Seppuku!"); }
```

Comment:
@RogerRiggs I suggest method name instead of `seppuku` to be `throwRuntimeException`?
And to remove the message from exception constructor.

Reply:
Both suggestions work for me.  Thanks